src/VisualStudio/Core/Def/Implementation/LanguageClient/LiveShareInProcLanguageClient.cs

```diff
@@ -42,6 +43,24 @@ internal class LiveShareInProcLanguageClient : AbstractInProcLanguageClient
         public override string Name => "Live Share C#/Visual Basic Language Server Client";
 
         protected internal override VSServerCapabilities GetCapabilities()
-            => _defaultCapabilitiesProvider.GetCapabilities();
+        {
+            var experimentationService = Workspace.Services.GetRequiredService<IExperimentationService>();
+            var isLspEditorEnabled = experimentationService.IsExperimentEnabled(VisualStudioWorkspaceContextService.LspEditorFeatureFlagName);
+            // If the preview feature flag to turn on the LSP editor in local scenarios is on, advertise no capabilities for this Live Share
+            // LSP server as LSP requests will be serviced by the AlwaysActiveInProcLanguageClient in both local and remote scenarios.
+            if (isLspEditorEnabled)
```

Comment:
I see ActivateAsync is annotated as being allowed to be null-returning; is that a better way to avoid spinning up a server that then won't do anything? The docs just say "or null if the language server cannot be started" which it's hard to tell if such a return is interpreted as a failure and might result in telemetry/error messages.

Reply:
Ah good point about the telemetry.  I checked and a failure event is recorded if we return null.
I talked to Miguel and the best supported way is to activate a server without any capabilities set.

Reply:
Sounds good.